notify/factory.go

```diff
@@ -0,0 +1,98 @@
+package notify
```

Comment:
I gotta admit, I'm not a fan of this pattern *at all*. I spent a big portion of my career writing this type of code in ruby and 10 years later I continue to find this hard to read/understand.

The layer of abstraction for the sake of saving an initialisation of an error seems too much.

That being said, I consider this a nit - if you all agree this is OK feel free to ignore my comment.

Reply:
Not sure to which line of code you refer to. Do you mean `wrap` function? If yes, I am not a fan of it either, this was made as a workaround for providing a validation of the configuration prior to saving it to the database.
Currently, there is a similar factory method for each receiver and it looks exactly the same. I just made it DRY-er.

Reply:
Ah, definitely, I think we should change it to the way it is implemented in Alertmanager!